src/App.js

```diff
@@ -1,16 +1,83 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
 import chatMessages from './data/messages.json';
+import SelectColor from './components/SelectColor';
+
+// iterate through chatMessages and create a list of senders
+const senders = [];
+for (const message of chatMessages) {
+  if (!senders.includes(message.sender)) {
+    senders.push(message.sender);
+  }
+}
+
+// assign first person in senders to local and second to remote (assumes only 2 senders and that the first sender is the local one, wouldn't work for group text!)
+const local = senders[0];
+const remote = senders[1];
+
+// maps through chatMessages and adds local/remote
+const chatMessagesWithLocation = chatMessages.map((message) => {
```

Comment:
👍 nicely done. Could rename the function to something like setMessageLocation(). While not a hard and fast rule, often times functions have verbs in the name to indicate what they do (for example: get/set/callApi).